app/services/reporting/agency_and_sp_report.rb

```diff
@@ -48,30 +48,36 @@ def agency_and_sp_emailable_report
     end
 
     def active_agencies
-      @active_agencies ||= Agreements::PartnerAccountStatus.find_by(name: 'active').
-        partner_accounts.
-        includes(:agency).
-        where('became_partner <= ?', report_date).
-        map(&:agency).
-        uniq
+      @active_agencies ||= Agency.joins(:partner_accounts).
+        where(partner_accounts: {
+          partner_account_status: Agreements::PartnerAccountStatus.find_by(name: 'active'),
+          became_partner: ..report_date,
+        }).
+        distinct
     end
 
     def service_providers
-      @service_providers ||= Reports::BaseReport.transaction_with_timeout do
```

Comment:
Discussed this in Slack a bit with @MrNagoo since I kind of suspect this query is the bigger issue. Currently it does a table scan of the entire `identities` table. Since it's a massive table, it will be very slow.

The query here is intending to return service providers that have users that have authorized the application and exclude the ones that do not. We don't have any existing index on `identities` that would be helpful for this. The proposed change uses `sp_return_logs` instead, which serves a similar purpose and has an index that can be used. It also splits it up so each individual service provider is queried to alleviate timeout issues.